[WD-19996] feat: added dropdown nav

muhammad-ali-pk opened the pull request:
## Done

- Added slide navigation
- Refactored build.js
- Use versioned_static resources

## QA

- Check out this feature branch
- Run the site using the command `docker compose up -d` and `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8017/manager
- Use medium/smaller screen size
- Verify the top navigation is working as expected

## Issue / Card

Fixes #[WD--19996](https://warthogs.atlassian.net/browse/WD-19996)

muhammad-ali-pk commented:
Note to reviewer: The navigation.js file content come straight from [Vanilla](https://vanillaframework.io/docs/patterns/navigation#sliding) and can be skipped in the code review process.

petesfrench commented:
@muhammad-ali-pk I think the [vanilla dropdown navigation](https://vanillaframework.io/docs/patterns/navigation#dropdown) is a better fit here. Sliding nav is designed to handled nested navigation items, which we don't have (and I suppose won't have) here

muhammad-ali-pk commented:
@petesfrench Thanks for the suggestion Pete. I was under the impression that the sliding navigation was a requirement as it was already being used as "p-navigation--sliding" though not working.
Since that's not a requirement, and I agree with you on dropdown nav being the better fit, I have changed it to the dropdown nav, and updated the navigation.js accordingly.

petesfrench commented:
When you click the search button, the active tab loses it's highlighting
before:
![image](https://github.com/user-attachments/assets/386b1632-9bcf-4fea-aec3-f2f42be4bb60)
after:
![image](https://github.com/user-attachments/assets/19d5c4c7-d746-477e-841e-4a86ef7bf957)

petesfrench reviewed (approved):
LGTM, a very minor qa comment. Feel free to merge once it is addressed

muhammad-ali-pk commented:
@petesfrench Thanks for pointing it out Pete. Fixed it.
